agent/engine/docker_task_engine.go

```diff
@@ -649,6 +649,19 @@ func (engine *DockerTaskEngine) deleteTask(task *apitask.Task) {
 		}
 	}
 
+	for _, container := range task.Containers {
+		for _, resource := range container.GetResources() {
+			err := resource.Cleanup()
+			if err != nil {
```

Comment:
I wonder if there's anything we could do apart from logging the clean-up failure. I realized this is what we currently have for task clean-up as well, but IMO the error should rather be surfaced to customers so that at least they could potentially take action such as manual clean-up, otherwise there may be resource leak without customer visibility. 

This ask probably does not fall in scope of this particular PR, but just want to bring it up for some thoughts, as we are making relevant changes.

Reply:
I took a look at current `deleteTask` and thought about this for a bit, Currently, `deleteTask` is called by `cleanUpTask` and then being monitored in `overseeTask`, I don't see any one in this call stack expects returns. I am wondering how would we want to emit the error apart from logging? I believe we don't want the actual task removal process to be aborted. Change log level to error makes some sense to me but from my understanding I think an error usually means we can't proceed with the current operation in an expected way (please let me know if I am wrong)